### PR TITLE
feat: `x/blocksdk`

### PR DESCRIPTION
**Pull request by aljo242**
This PR adds the `x/blocksdk` module.

This PR does not add the integration into the lane-setting code of the SDK.  That work will be done in a follow up PR while this is finalized.

### CODE REVIEW

**Comment by aljo242** [comment]
Note: files modifed in `abci`, `adapters`, and `block` are just from running the formatter.

File: block/base/priority_nonce.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ import (
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkmempool "github.com/cosmos/cosmos-sdk/types/mempool"
+
```
Comment (nivasan1) [inline block/base/priority_nonce.go:24]: kek

File: proto/sdk/blocksdk/v1/blocksdk.proto [hunk 2]
```diff
@@ -0,0 +1,18 @@
+syntax = "proto3";
+package sdk.blocksdk.v1;
+
+import "gogoproto/gogo.proto";
+import "amino/amino.proto";
+import "cosmos_proto/cosmos.proto";
+
+option go_package = "github.com/skip-mev/block-sdk/x/blocksdk/types";
+
```
Comment (nivasan1) [inline proto/sdk/blocksdk/v1/blocksdk.proto:9]: nit: docs

Comment (nivasan1) [reply proto/sdk/blocksdk/v1/blocksdk.proto:9]: As a rule of thumb, we like to provide docs on our proto obj, if you write them in the form of [godocs](https://go.dev/blog/godoc), the documentation will appear in the generated objects

File: proto/sdk/blocksdk/v1/query.proto [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+option go_package = "github.com/skip-mev/block-sdk/x/blocksdk/types";
+
+import "google/api/annotations.proto";
+import "gogoproto/gogo.proto";
+import "cosmos/query/v1/query.proto";
+import "sdk/blocksdk/v1/blocksdk.proto";
+
+// Query defines the x/blocksdk querier service.
+service Query {
+  rpc Lane(QueryLaneRequest) returns (QueryLaneResponse) {
```
[the remark this answers is not in the document]
Comment (nivasan1) [reply proto/sdk/blocksdk/v1/query.proto:13]: asking to learn, not suggesting a change

File: block/base/priority_nonce.go (hunk 1 above)
Comment (aljo242) [reply block/base/priority_nonce.go:24]: I'm just running `make format`

File: proto/sdk/blocksdk/v1/query.proto (hunk 3 above)
[the remark this answers is not in the document]
Comment (aljo242) [reply proto/sdk/blocksdk/v1/query.proto:13]: This is sort of the typical [API](https://github.com/cosmos/cosmos-sdk/blob/630836e6edd5987b5f1e00556ca6f7921e1902fb/proto/cosmos/bank/v1beta1/query.proto#L18) I see people exposing in cosmos.  
- single query by some identifier
- a list of all objects

Maybe we can add pagination to the `Lanes()` query?

File: proto/sdk/blocksdk/v1/tx.proto [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import "cosmos_proto/cosmos.proto";
+import "google/api/annotations.proto";
+import "amino/amino.proto";
+import "sdk/blocksdk/v1/blocksdk.proto";
+
+// Msg defines the x/blocksdk Msg service.
+service Msg {
+  option (cosmos.msg.v1.service) = true;
+
+  rpc RegisterLane(MsgRegisterLane) returns (MsgRegisterLaneResponse) {
```
Comment (nivasan1) [inline proto/sdk/blocksdk/v1/tx.proto:17]: WB Deleting a Lane? If we have a `RegisterLane` flow, I'd imagine we shld also introduce a `RemoveLane` flow as well, no? However, in both cases (`Register`, `Delete`) chain devs will have to configure the lanes in the app anyway, so I'm not sure what invariant / feature these methods (`Register`, `Delete`) are intended to introduce?

Comment (nivasan1) [reply proto/sdk/blocksdk/v1/tx.proto:17]: Then again, I may be misunderstanding the expected wiring logic

File: tests/app/config.go [hunk 5]
```diff
@@ -66,8 +66,11 @@ import (
 	_ "github.com/cosmos/cosmos-sdk/x/staking"      // import for side-effects
 
 	auctionmodulev1 "github.com/skip-mev/block-sdk/api/sdk/auction/module/v1"
+	blocksdkmodulev1 "github.com/skip-mev/block-sdk/api/sdk/blocksdk/module/v1"
 	_ "github.com/skip-mev/block-sdk/x/auction" // import for side-effects
 	auctiontypes "github.com/skip-mev/block-sdk/x/auction/types"
+	_ "github.com/skip-mev/block-sdk/x/blocksdk" // import for side-effects
```
Comment (nivasan1) [inline tests/app/config.go:72]: What side effects are necessary? Usually, we do _ imports if there is a useful `init()` fn in the package

File: tests/app/config.go [hunk 6]
```diff
@@ -247,6 +251,10 @@ var (
 				Name:   auctiontypes.ModuleName,
 				Config: appconfig.WrapAny(&auctionmodulev1.Module{}),
 			},
+			{
+				Name:   blocksdktypes.ModuleName,
+				Config: appconfig.WrapAny(&blocksdkmodulev1.Module{}),
```
Comment (nivasan1) [inline tests/app/config.go:256]: dep-inject wizard 💯

File: x/blocksdk/keeper/keeper.go [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/skip-mev/block-sdk/x/blocksdk/types"
+)
+
+type Keeper struct {
+	cdc      codec.BinaryCodec
+	storeKey storetypes.StoreKey
+
+	// The address that is capable of executing a MsgUpdateParams message.
```
Comment (nivasan1) [inline x/blocksdk/keeper/keeper.go:20]: `MsgUpdateParams` is not defined on this module, no? Do you mean `UpdateLane` / `RegisterLane`

File: x/blocksdk/keeper/keeper.go [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+	// Typically this will be the governance module's address.
+	authority string
+}
+
+// NewKeeper is a wrapper around NewKeeperWithRewardsAddressProvider for backwards compatibility.
```
Comment (nivasan1) [inline x/blocksdk/keeper/keeper.go:25]: nit: update docs

File: x/blocksdk/keeper/keeper_test.go [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+)
+
+type KeeperTestSuite struct {
+	suite.Suite
+
+	blocksdKeeper    keeper.Keeper
+	bankKeeper       *mocks.BankKeeper
+	accountKeeper    *mocks.AccountKeeper
+	distrKeeper      *mocks.DistributionKeeper
+	stakingKeeper    *mocks.StakingKeeper
```
Comment (nivasan1) [inline x/blocksdk/keeper/keeper_test.go:27]: nit: Why do we need bank, account, distr, stakingKeeper for this suite?

File: x/blocksdk/module.go [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+}
+
+type Inputs struct {
+	depinject.In
+
+	Config *modulev1.Module
+	Cdc    codec.Codec
+	Key    *storetypes.KVStoreKey
+
+	AccountKeeper      types.AccountKeeper
```
Comment (nivasan1) [inline x/blocksdk/module.go:161]: Do we need these keepers?

File: x/blocksdk/types/keys.go [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+	// RouterKey is the message route for the blocksdk module
+	RouterKey = ModuleName
+
+	// QuerierRoute is the querier route for the blocksdk module
+	QuerierRoute = ModuleName
+)
+
+const (
+	prefixLanes = iota + 1
```
Comment (nivasan1) [inline x/blocksdk/types/keys.go:18]: nit: iota

Comment (nivasan1) [reply x/blocksdk/types/keys.go:18]: nit: Or if you'd want this to be 1, j `prefixLanes = 1`

File: x/blocksdk/types/msgs.go [hunk 12]
```diff
@@ -0,0 +1,71 @@
+package types
+
```
Comment (nivasan1) [inline x/blocksdk/types/msgs.go:2]: nit: test-cases?

Comment (nivasan1) [reply x/blocksdk/types/msgs.go:2]: Or do we test these cases in the `KeeperTestSuite`

**Review by nivasan1 — COMMENTED** [review]
Left a few nits + a question on the `Msg` service

File: x/blocksdk/types/msgs.go (hunk 12 above)
Comment (aljo242) [reply x/blocksdk/types/msgs.go:2]: We do test these in `KeeperTestSuite` but I think we should have `types` level tests too

**Pull request closed without merging**